pepper-apis/Dockerfile-backend

```diff
@@ -32,14 +31,11 @@ ENV PUBSUB_EMULATOR_HOST "localhost:8442"
 RUN ["echo","Frontloading maven dependencies..."]
 RUN mvn $SET_LOCAL_M2_REPO --batch-mode dependency:go-offline install surefire-report:report-only \
   || echo "warning: failed to cache pepper-apis maven dependencies"
-RUN mvn $SET_LOCAL_M2_REPO -f smoketest-pom.xml --batch-mode dependency:go-offline install \
-  || echo "warning: failed to cache smoketest maven dependencies"
 RUN mvn $SET_LOCAL_M2_REPO -f housekeeping-pom.xml --batch-mode dependency:go-offline install \
   || echo "warning: failed to cache housekeeping maven dependencies"
 
 COPY src/ /app/src
 RUN mvn $SET_LOCAL_M2_REPO -o --batch-mode -DskipTests=true clean install package || echo "warning: failed to build pepper-apis"
-RUN mvn $SET_LOCAL_M2_REPO -o --batch-mode -DskipTests=true --file smoketest-pom.xml package || echo "warning: failed to build pepper-apis/smoketest"
```

Comment:
Let's wait to remove this until all environments have the backend on GAE.

This makes me wonder about something, though: with manual scaling, if the backend falls over during a liquibase migration, nothing will restart it, right?  We should probably figure out liveness checks and readiness checks for GAE backends.

Reply:
For manual scaling, I think it depends. If the app doesn't respond to the `/_ah/start` call, I believe GAE will reboot it. Although the wait time seems to be around 10 minutes, as evident from the time when constant Housekeeping reboots was driving us bonkers. In Housekeeping's setup, the very last thing we do is respond to GAE. This is after the liquibase migration runs. So if things failed there, I suspect we won't respond to GAE and thus will get rebooted. Not too familiar with study-manager, but in that case, either liquibase migration throws an exception and halts startup or we run out of time and respond GAE handler returns a `503`. Either way, I think it will get rebooted by GAE.

I think the bigger question is, if and how we will get notified of that failure so we can fix it. Because rebooting it probably won't help make it any better.